src/build/cargo.rs

```diff
@@ -299,20 +308,29 @@ impl Executor for RlsExecutor {
             }
         }
 
-        if self.workspace_mode {
-            let output = cmd.output().expect("Couldn't execute rustc");
-            let mut stderr_json_msg = convert_message_to_json_strings(output.stderr);
-            self.compiler_messages.lock().unwrap().append(&mut stderr_json_msg);
-        } else {
-            cmd.status().expect("Couldn't execute rustc");
-        }
-
         // Finally, store the modified cargo-generated args/envs for future rustc calls
         args.insert(0, rustc_exe);
         let mut compilation_cx = self.compilation_cx.lock().unwrap();
         compilation_cx.args = args;
         compilation_cx.envs = cargo_cmd.get_envs().clone();
 
+        if self.workspace_mode {
+            let args = &compilation_cx.args;
+            let envs = &compilation_cx.envs;
```

Comment:
I don't see how these ever get populated since we never do a Cargo run without running rustc directly in workspace mode? How does this work?